pkg/handlers/generic/lifecycle/serviceloadbalancer/metallb/handler.go

```diff
@@ -152,5 +158,77 @@ func (n *MetalLB) Apply(
 		return fmt.Errorf("failed to apply MetalLB installation HelmChartProxy: %w", err)
 	}
 
+	if err := wait.ForObject(
+		ctx,
+		wait.ForObjectInput[*caaphv1.HelmChartProxy]{
+			Reader: n.client,
+			Target: hcp.DeepCopy(),
+			Check: func(_ context.Context, obj *caaphv1.HelmChartProxy) (bool, error) {
+				for _, c := range obj.GetConditions() {
+					if c.Type == caaphv1.HelmReleaseProxiesReadyCondition && c.Status == corev1.ConditionTrue {
+						return true, nil
+					}
+				}
+				return false, nil
+			},
+			Interval: 5 * time.Second,
+			Timeout:  10 * time.Second,
+		},
+	); err != nil {
+		return fmt.Errorf("failed to wait for MetalLB to deploy: %w", err)
+	}
+
+	log.Info(
+		fmt.Sprintf("Applying MetalLB configuration to cluster %s",
+			ctrlclient.ObjectKeyFromObject(cluster),
+		),
+	)
+
+	cos, err := configurationObjects(&configurationInput{
+		name:          defaultHelmReleaseName,
+		namespace:     defaultHelmReleaseNamespace,
+		addressRanges: slb.Configuration.AddressRanges,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to generate MetalLB configuration: %w", err)
+	}
+
+	var applyErr error
+	if waitErr := kwait.PollUntilContextTimeout(
+		ctx,
+		5*time.Second,
+		10*time.Second,
```

Comment:
Just checking that this is the intended timeout? During testing we were seeing that it backed off pretty aggressively and CAPI didn't call the hook for a few mins.